### PR TITLE
Added dark mode switch listener on Windows and Linux

### PR DESCRIPTION
**Pull request by TransparentLC**
Implemented with WIN32 API on Windows and `gsettings monitor org.gnome.desktop.interface gtk-theme` on Linux. See my comment in #14 for details.

### CODE REVIEW

**Comment by albertosottile** [comment]
Hi, thank you for your contribution. The implementation is nice and clean. However, the rationale behind `darkdetect` was to have a lightweight pure-Python package, with no non-standard dependencies, that could be embedded in more complex Python projects (e.g. [Syncplay](https://github.com/syncplay/syncplay)). I feel that introducing platform specific Python wheels would break this philosophy. 

At the moment, as I mentioned in #14, this is a task that I see more suitable for the GUI library. Nevertheless, if you could manage to reimplement this code through `ctypes`, then I would definitely accept it as it would eliminate all the previously mentioned drawbacks, while still providing this valuable feature.

**Comment by TransparentLC** [comment]
I have reimplemented the listener by ctypes. Feel free to try it out!

File: README.md [hunk 1]
```diff
@@ -25,6 +25,18 @@ False
 ```
 It's that easy.
 
+You can create a dark mode switch listener daemon thread with `darkdetect.listener` and pass a callback function. The function will be called with string "Dark" or "Light" when the OS switches the dark mode setting.
+
+``` python
+import darkdetect, threading
```
Comment (albertosottile) [inline README.md:31]: Please separate the two imports on two lines, as in 
```python
import threading
import darkdetect
```

File: darkdetect/_dummy.py [hunk 2]
```diff
@@ -12,3 +14,6 @@ def isDark():
     
 def isLight():
     return None
+
+def listener(callback: typing.Callable[[str], None]) -> None:
+    raise NotImplementedError()
```
Comment (albertosottile) [inline darkdetect/_dummy.py:19]: Please add a new line at the end of this file.

File: darkdetect/_mac_detect.py [hunk 3]
```diff
@@ -68,3 +69,6 @@ def isDark():
 
 def isLight():
     return theme() == 'Light'
+
+def listener(callback: typing.Callable[[str], None]) -> None:
+    raise NotImplementedError()
```
Comment (albertosottile) [inline darkdetect/_mac_detect.py:74]: Please add a new line at the end of this file.

**Review by albertosottile — CHANGES_REQUESTED** [review]
Thank you very much for this contribution. I tested it on Windows and the code LGTM, I only have a few minor comments.

**Review by albertosottile — APPROVED** [review]

**Comment by albertosottile** [comment]
Thank you again for this. I will try to create a new release later today.

**Pull request merged**